### ISSUE TITLE
Macros rewrite Math.PI -> ad.scalar.PI which is undefined

### ISSUE THREAD

**Issue by null-a** [dritchie/adnn#1]
In WebPPL, I've done something like this:

```
// Re-export Math constants etc.
Object.getOwnPropertyNames(Math).forEach(function(p) {
  if (!ad.scalar.hasOwnProperty(p)) {
    ad.scalar[p] = Math[p];
  }
});
```

(Which is similar to what happens in ad.js.)

Would you accept a PR adding something similar to `functions.js`, or would you prefer to handle it in some other way?

**Comment by dritchie** [comment]
Sure, that's fine by me.

**Comment by dritchie** [comment]
Closed by 5388de2
